src/olympia/discovery/tests/test_admin.py

```diff
@@ -1089,15 +1093,16 @@ def test_can_edit_with_discovery_edit_permission(self):
         content = response.content.decode('utf-8')
         assert 'Recommended extensions' in content
 
+        clean_criteria_mock.return_value = {
+            'title': 'Popular extensions',
+            'shelf_type': 'extension',
+            'criteria': '?recommended=true&sort=users&type=extension'
+        }
```

Comment:
I don't think you need this `return_value`?  The `clean_criteria` method wouldn't usually return anything - it just raises exceptions for problems and returns nothing otherwise.

This is the kind of return you'd expect from the responses.add though.

Reply:
The `clean_criteria` method returns the criteria. When I removed `return criteria`, the `criteria` field errors to `This field cannot be null` when there is a valid criteria in the field. I had it as just a `clean` method before which did not return anything, but I anticipate adding a `clean_footer_pathname` later and did not one huge `clean` method. Thoughts?